docs/rfc/rfc-020-onboarding-projects.rst

```diff
@@ -0,0 +1,235 @@
+=======================================
+RFC 020: Tendermint Onboarding Projects
+=======================================
+
+.. contents::
+   :backlinks: none
+
+Changelog
+---------
+
+- 2022-03-30: Initial draft. (@tychoish)
+- 2022-04-25: Imported document to tendermint repository. (@tychoish)
+
+Overview
+--------
+
+This document describes a collection of projects that might be good for new
+engineers joining the Tendermint Core team. These projects mostly describe
+features that we'd be very excited to see land in the code base, but that are
+intentionally outside of the critical path of a release on the roadmap, and
+have the following properties that we think make good on-boarding projects:
+
+- require relatively little context for the project or its history beyond a
+  more isolated area of the code.
+
+- provide exposure to different areas of the codebase, so new team members
+  will have reason to explore the code base and gain expeirence with more than
+  one area of the system.
+
+- be of moderate size, thereby avoiding making small trivial or mechanical changes
+  while also avoiding large intractable changes without natural touchpoints or
+  check-ins.
+  
+Projects
+--------
+
+Before diving into one of these projects, definitely check in with your
```

Comment:
It might make more sense to address this project to the buddy, rather than (or in addition to) the new person onboarding. I agree with what you wrote here, and it implies a certain amount of preparation on the part of us existing team members.

I would foresee it going something like:

- Buddy: "Read through this list and see if any of these things interest you."
- Onboarder: "I think X and Y are interesting."
- Buddy: (goes off and writes up some preliminary orientation material)

We could also try to do that third step ahead of time for all of these, but I find in practice that kind of work gets stale quickly. Better, I think, to do it on-demand. But it does imply a certain responsibility we have to budget for.

Reply:
I agree with your view of how the process might go. I'm tweaking the text here.

I think that we might walk away from an interview with an idea of what people might want to work on, and could do the design/planning step JIT style.